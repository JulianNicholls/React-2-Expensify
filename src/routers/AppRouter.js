import React from 'react';
import { BrowserRouter, Route, Switch, Link } from 'react-router-dom';

import Header from '../components/Header';
import LoginPage from '../components/LoginPage';
import DashboardPage from '../components/DashboardPage';
import AddExpensePage from '../components/AddExpensePage';
import EditExpensePage from '../components/EditExpensePage';

export const NotFoundPage = () => (
  <React.Fragment>
    <h1>404</h1>
    <p>That page was not found.</p>
    <p>
      <Link to="/">Back to bed, back to reality</Link>
    </p>
  </React.Fragment>
);

const AppRouter = () => (
  <BrowserRouter>
    <div>
      <Header />
      <Switch>
        <Route path="/" exact component={LoginPage} />
        <Route path="/dashboard" component={DashboardPage} />
        <Route path="/create" component={AddExpensePage} />
        <Route path="/edit/:id" component={EditExpensePage} />

        <Route component={NotFoundPage} />
      </Switch>
    </div>
  </BrowserRouter>
);

export default AppRouter;
