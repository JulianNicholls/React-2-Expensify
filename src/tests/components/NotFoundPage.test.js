import React from 'react';
import { shallow } from 'enzyme';

import { NotFoundPage } from '../../routers/AppRouter.js';

describe('NotFoundPage component', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<NotFoundPage />);

    expect(wrapper).toMatchSnapshot();
  });
});
