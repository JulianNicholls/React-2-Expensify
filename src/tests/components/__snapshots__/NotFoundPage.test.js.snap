// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundPage component should render correctly 1`] = `
<React.Fragment>
  <h1>
    404
  </h1>
  <p>
    That page was not found.
  </p>
  <p>
    <Link
      replace={false}
      to="/"
    >
      Back to bed, back to reality
    </Link>
  </p>
</React.Fragment>
`;
